src/v/cluster/rm_stm.h

```diff
@@ -230,7 +233,8 @@ class rm_stm final : public raft::persisted_stm<> {
     ss::future<> do_remove_persistent_state();
     ss::future<fragmented_vector<tx::tx_range>>
       do_aborted_transactions(model::offset, model::offset);
-    tx::producer_ptr maybe_create_producer(model::producer_identity);
+    std::pair<tx::producer_ptr, bool>
```

Comment:
nit: an explicity type or an alias for `ss::bool_class` would make it more readable